lib/mongoid/criteria/queryable/extensions/boolean.rb

```diff
@@ -20,7 +20,8 @@ module ClassMethods
             # @return [ true, false ] The boolean value.
             def evolve(object)
               __evolve__(object) do |obj|
-                obj.to_s =~ (/\A(true|t|yes|y|on|1|1.0)\z/i) ? true : false
+                res = mongoize(object)
+                res.nil? ? object : res
```

Comment:
Is there a behavior change here? If not I would prefer this hunk reverted here and done as part of evolution PR.

Reply:
This logic could still be copied into mongoize for now.